webapp/client/src/components/FormFieldRadio.js

```diff
@@ -0,0 +1,132 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabelForSeparatedFields from "./FieldLabelForSeparatedFields";
+import { optionsPropType } from "../lib/propTypes";
+
+const Radio = styled.div`
+  input[type="radio"] {
+    opacity: 0;
+    position: absolute;
+  }
+
+  input[type="radio"] + label {
+    display: flex;
+    padding-left: 0;
+    padding-top: 0;
+    font-size: var(--text-medium);
+  }
+
+  input[type="radio"] + label::before {
+    content: "";
+    display: inline-block;
+    vertical-align: bottom;
+    width: 30px;
+    height: 30px;
+    min-width: 30px;
+    margin-right: 12px;
+    background: url("/icons/radio_button_default.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked + label::before {
+    background: url("/icons/radio_button_checked.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked + label:hover::before {
+    background: url("/icons/radio_button_checked_hover.svg") no-repeat center;
+  }
+
+  input[type="radio"]:not(:checked):focus + label::before {
+    background: url("/icons/radio_button_focus.svg") no-repeat center;
+  }
+
+  input[type="radio"]:checked:focus + label::before {
+    background: url("/icons/radio_button_checked_focus.svg") no-repeat center;
+  }
+
+  .radio-button-list {
+    padding-top: var(--spacing-02);
+  }
+`;
+
+function FormFieldRadio({
+  fieldName,
+  fieldId,
+  options,
+  value,
+  required,
+  autofocus,
+  label,
+  details,
+  errors,
+  onChangeHandler,
+}) {
+  const [selectedValue, setSelectedValue] = useState(value);
+
+  const toggleRadioButton = (event) => {
+    setSelectedValue(event.target.value);
+  };
+
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+      }}
+      hideLabel
+      render={() => (
+        <Radio>
+          <fieldset id={fieldId} name={fieldId}>
+            <FieldLabelForSeparatedFields {...{ label, fieldId, details }} />
+            <div className="radio-button-list">
+              {options.map((option, i) => [
```

Comment:
Ooooh! Sorry, _now_ I get it! This isn't a component for a radio button, it's a component for a radio group! Can you change the name to FormFieldRadioGroup (I'm taking the name "radio group" from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio))?

I also think what you call `options` should be called `radioButtons` - options is very confusing because there's an element with that name!

Reply:
Iiiinteresting. I've been using it all wrong the whole time. 🤔

Reply:
I just had another look at the page I linked and they do use the word options as well, so maybe I'm wrong: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio